test/test_main.py

```diff
@@ -60,3 +60,34 @@ def test_main(self):
             [{'student_uniqname': 'nlongbottom', 'score': 500.0, 'transmitted': True}]
         )
         self.assertEqual(len(mail.outbox), 1)
+
+    def test_main_does_not_send_email_when_no_transmission_activity(self):
+        """
+        Function main does not send email when there are no successful or failed transmissions.
+        (Note this also implies that there were no new submissions found.)
+        """
+        with patch('pe.orchestration.api_call_with_retries', autospec=True) as mock_get:
+            mock_get.return_value = MagicMock(spec=Response, status_code=200, text=json.dumps([]))
+            main(self.api_handler)
+
+        dada_report: Report = Report.objects.get(id=3)
+        new_submissions_qs: QuerySet = dada_report.exams.first().submissions.all()
+        self.assertFalse(new_submissions_qs.exists())
+        self.assertEqual(len(mail.outbox), 0)
+
+    def test_main_sends_email_when_only_transmission_failures(self):
+        """
+        Function main still sends email when there are failed transmissions.
+        """
+        with patch('pe.orchestration.api_call_with_retries', autospec=True) as mock_get:
+            with patch.object(ApiUtil, 'api_call', autospec=True) as mock_send:
+                mock_get.return_value = MagicMock(
+                    spec=Response, status_code=200, text=json.dumps(self.canvas_dada_place_subs)
+                )
+                mock_send.return_value = MagicMock(spec=Response, status_code=500, text=json.dumps({}))
```

Comment:
can add a comment line here for "mock failed submission with status code 500"

Reply:
Hm, I'll think about it. Not sure it adds that much, since the docstring and test name already kinda flag that that's what is going to happen.

Reply:
I'm gonna skip this.